integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -267,9 +266,8 @@ object CarbonDataRDDFactory {
             throw new Exception("Exception in compaction " + exception.getMessage)
           }
         } finally {
-          executor.shutdownNow()
           try {
-            compactor.deletePartialLoadsInCompaction()
```

Comment:
a) When compaction retries, it uses the same segment ID, if stale files are not cleaned. It gives duplicate data.
So, before this change, we need #3934 to be merged which can use a unique segment id for compaction retry.

b) please check and move the logic of deletePartialLoadsInCompaction in clean files command, instead of permanently removing it. If the clean files don't have this logic, it may not able to clean stale files.

c) Also if the purpose of this PR is to avoid accidental data loss. you need to handle `cleanStaleDeltaFiles` in `CarbonUpdateUtil.java` and also identify other places. Just handling in few place will not guarantee that we cannot have data loss.

Reply:
(a). My PR will dependent on #3934 , if it merged first we don't need to worry about the same segment ID in compaction situations. So it the deletePartialLoadsInCompaction could be deleted. 

(b). In this PR's design, the stale files would not be deleted automatically. Users need to call cleanFiles function to clean them besides PR #3917 can handle the clean files function.

(c). More places of calling claenStaleFiles have been removed.

Reply:
a) root cause of stale files:  It uses listFiles to collect index files when writing the segment file, so it will add stale index file names into segment file. 
  it is hard to add a unique id, so we change the plan to avoid to listFiles during wirte segment file.

Reply:
@QiangCai how its handled now, without list files? why cant we do list files with the timestamp filter, which is load timestamp/fact timestamp, we can get from load model or somewhere right?

Reply:
> We cannot remove the clean stale files in case of IUD and wait for clean files command to clean them, we should immediately clean the stale ones in the respective command itself, as there will be chances of extra data or data inconsistency.

Reply:
@akashrn5 After we avoid using listFiles during loading,  the stale segment (for example 0.1) will not impact data consistency. If the stale segment has stale index files and data files,  we will not add them to segment file.

Reply:
@ajantha-bhat after the merge of #3978 , cleanStaleDeltaFiles will only be called when there are exceptions during the update process. And it will only delete the delta and index file created by the exception update.